doc/build_system.dox

```diff
@@ -0,0 +1,549 @@
+  are very few of these, mostly for FreeRTOS and some used for patching broken
+  libc and middleware distributions.
+
+- {@code src} The source code of OpenMRN core libraries. These are:
+
+  - {@code console} A telnet-like server that you can use to present a
+    commandline console. It can operate via network socket or serial
+    connection.
+
+  - {@code dcc} Declarations and classes for implementing and representing DCC
+    protocol; used by command stations, mobile decoders and RailCom detectors.
+
+  - {@code executor} Core library for co-routine based collaborative
+    multi-tasking. Allows time division multi-tasking on a single thread,
+    including running a large number of state machines interacting with message
+    queues to for protocol stacks.
+
+  - {@code freertos_drivers} Posix-compatible hardware abstraction layer
+    running on FreeRTOS, with driver implementations for a number of popular
+    microcontrollers' common hardware (such as UART, USB, and CAN-bus drivers).
+
+    - {@code common} Base classes for drivers and posix device abstraction.
+    - {@code esp8266} Implementations for the Espressif ESP8266 MCU.
+    - {@code net_cc32xx} Socket layer for the Texas Instruments CC32xx Wifi
+MCUs.
+    - {@code net_freertos_tcp} Socket layer using FreeRTOSPlus TCP.
+    - {@code nxp} Drivers for the NXP LPC17xx/40xx and 11Cxx MCUs.
+    - {@code pic32mx} Drivers for the Microchip PIC32 series, a MIPS4K based
+MCU.
+
+    - {@code st} Drivers for various stm32 MCUs, including F0, F1, F4 lines,
+  which are low-cost Cortex-M0 / M3 / M4 based MCUs.
+
+    - {@code ti} General drivers (CAN, UART, USB, GPIO) for Texas Instruments
+      Tiva and CC32xx MCUs.
+
+  - {@code openlcb} LCC / OpenLCB protocol stack. Most of this protocol stack
+    is implemented using protocol state flows and can operate on
+    single-threaded environments using the Executor framework.
+
+  - {@code os} OS abstraction layer with common concepts cross Linux, Mac,
+    FreeRTOS, and to some extent windows. The concepts include threads,
+    mutexes, sempahores, timers, support for select, GPIOs, mDNS.
+
+  - {@code utils} A large number of diverse utility classes for data
+    structures, buffers & queues, logging, testing, printing and formatting
+    data, creating and using streams and sockets. Helper classes for
+    implementing data hubs / repeaters, including tying them to various device
+    drivers such as CAN-bus or USB.
+
+  - {@code withrottle} Implementation for JMRI's wireless throttle protocol.\
+
+- {@code targets} All core targets. These are listed in importance order, not
+alphabetically:
+
+  - {@code cov} Target for compiling and running unit tests (on the host) and
+    test coverage generation.
+
+  - {@code linux.x86} Host target using 'g++' as the compiler. This can be used
+    to compile for the host OS under linux. It also work on a Raspberry Pi.
```

Comment:
Done, thanks. Next time if you want to write a shorter comment, just say
work->works